### PR TITLE
[BYTEMAN-428] MethodExpression#getPathCount quickfix

### PR DESCRIPTION
**Pull request by tadamski**
https://issues.redhat.com/browse/BYTEMAN-428

### CODE REVIEW

**Comment by adinn** [comment]
Hi Tomasz. Yes, there is definitely a bug here and it appears to have been there since 2009! Well done for finding it.

**Review by adinn — CHANGES_REQUESTED** [review]

File: agent/src/main/java/org/jboss/byteman/rule/expression/MethodExpression.java [hunk 1]
```diff
@@ -718,10 +718,12 @@ public int getPathCount(String name)
...
             int idx;
             for (idx = 0; idx < pathList.length; idx++) {
-                charCount += (dotExtra + pathList[idx].length());
+                charCount += pathList[idx].length();
+                if (name.charAt(charCount) == '.') {
+                    charCount++;
+                }
                 if (charCount > charMax) {
                     break;
                 }
```
[the remark this answers is not in the document]
Comment (tadamski) [reply agent/src/main/java/org/jboss/byteman/rule/expression/MethodExpression.java:727]: @adinn Thanks for the explanation. tbh all the corner cases that this method takes care of weren't obvious for me  when I was trying to think of the fix. I have update the PR according to your suggestion.

**Comment by adinn** [comment]
Fix for main branch -- still needs backporting to 3.X branch

**Comment by adinn** [comment]
Fix also pushed to 3.X branch

**Pull request merged**